src/gui/ogr/qgsogrhelperfunctions.h

```diff
@@ -19,18 +19,42 @@
 #include <QString>
 #include "qgis_gui.h"
 
+class QWidget;
+class QgsGdalOption;
+
 #define SIP_NO_FILE
 
 /**
- * CreateDatabaseURI
- * \brief Create database uri from connection parameters
- * \note not available in python bindings
+ * \ingroup gui
+ * \class QgsGdalGuiUtils
+ * \brief Utility functions for working with GDAL in GUI classes.
+ *
+ * \note not available in Python bindings
+ * \since QGIS 3.40
  */
-QString GUI_EXPORT createDatabaseURI( const QString &connectionType, const QString &host, const QString &database, QString port, const QString &configId, QString username, QString password, bool expandAuthConfig = false );
+class GUI_EXPORT QgsGdalGuiUtils
```

Comment:
shouldn't the file be renamed to qgsgdalguiutils.h for consistency?

Reply:
Yes, it should! I'll do that in a follow up PR I've got in progress to avoid merge conflicts.

Reply:
Done in https://github.com/qgis/QGIS/pull/57868